sale_elaboration/models/stock_picking.py

```diff
@@ -0,0 +1,25 @@
+# Copyright 2018 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, models
+from collections import defaultdict
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    @api.multi
+    def action_done(self):
+        res = super(StockPicking, self).action_done()
+        elaboration_product = defaultdict(lambda: 0.0)
+        for pick in self.filtered(lambda x: x.picking_type_code == 'outgoing'):
+            elaboration_lines = pick.move_lines.filtered(
```

Comment:
It's better to take here directly sale.order.line records:
```python
pick.mapped('move_lines.sale_line_id').filtered(
    ....
```

Reply:
I do not understand... i want stock move records not sale order lines records...

Reply:
OK, got it.